hazelcast/src/main/java/com/hazelcast/map/impl/mapstore/writebehind/DefaultWriteBehindProcessor.java

```diff
@@ -346,21 +346,14 @@ public void callAfterStoreListeners(Collection<DelayedEntry> entries) {
      * @return not-stored entries per partition.
      */
     private Map<Integer, List<DelayedEntry>> doStoreUsingBatchSize(List<DelayedEntry> sortedDelayedEntries) {
-        final Map<Integer, List<DelayedEntry>> failsPerPartition = new HashMap<Integer, List<DelayedEntry>>();
+        Map<Integer, List<DelayedEntry>> failsPerPartition = new HashMap<Integer, List<DelayedEntry>>();
         int page = 0;
         List<DelayedEntry> delayedEntryList;
         while ((delayedEntryList = getBatchChunk(sortedDelayedEntries, writeBatchSize, page++)) != null) {
-            final Map<Integer, List<DelayedEntry>> fails = processInternal(delayedEntryList);
-            final Set<Map.Entry<Integer, List<DelayedEntry>>> entries = fails.entrySet();
+            Map<Integer, List<DelayedEntry>> fails = processInternal(delayedEntryList);
+            Set<Map.Entry<Integer, List<DelayedEntry>>> entries = fails.entrySet();
             for (Map.Entry<Integer, List<DelayedEntry>> entry : entries) {
-                final Integer partitionId = entry.getKey();
-                final List<DelayedEntry> tmpFailList = entry.getValue();
-                List<DelayedEntry> failList = failsPerPartition.get(partitionId);
-                if (failList == null || failList.isEmpty()) {
-                    failsPerPartition.put(partitionId, tmpFailList);
-                    failList = failsPerPartition.get(partitionId);
-                }
-                failList.addAll(tmpFailList);
```

Comment:
this is the line with the issue. put `tmpFailList` to map then get that list back and then add that list again to the map.